prov/verbs/src/ep_rdm/verbs_rdm_cm.c

```diff
@@ -815,72 +817,51 @@ fi_ibv_rdm_process_event(struct rdma_cm_event *event, struct fi_ibv_rdm_ep *ep)
 
 ssize_t fi_ibv_rdm_cm_progress(struct fi_ibv_rdm_ep *ep)
 {
-	struct rdma_cm_event *event = NULL;
-	void *data = NULL;
-	ssize_t ret = FI_SUCCESS;
+	ssize_t ret;
+
+	do {
+		struct rdma_cm_event event_copy;
+		struct fi_conn_priv_params priv;
+		struct rdma_cm_event *event;
+
+		if (rdma_get_cm_event(ep->domain->rdm_cm->ec, &event)) {
+			if (errno == EAGAIN) {
+				usleep(ep->domain->rdm_cm->cm_progress_timeout);
+				ret = FI_SUCCESS;
+				break;
+			}
 
-	if (rdma_get_cm_event(ep->domain->rdm_cm->ec, &event)) {
-		if(errno == EAGAIN) {
-			usleep(ep->domain->rdm_cm->cm_progress_timeout);
-			return FI_SUCCESS;
-		} else {
 			VERBS_INFO_ERRNO(FI_LOG_AV,
 					 "rdma_get_cm_event failed\n", errno);
 			ret = -errno;
+			break;
 		}
-	}
 
-	while (ret == FI_SUCCESS && event) {
-		pthread_mutex_lock(&ep->domain->rdm_cm->cm_lock);
-
-		struct rdma_cm_event event_copy;
 		memcpy(&event_copy, event, sizeof(*event));
 		if (event->param.conn.private_data_len) {
-			data = malloc(event->param.conn.private_data_len);
-			if (!data) {
-				pthread_mutex_unlock(&ep->domain->rdm_cm->cm_lock);
-				ret = -FI_ENOMEM;
-				break;
-			}
-			memcpy(data, event->param.conn.private_data,
-				      event->param.conn.private_data_len);
-			event_copy.param.conn.private_data = data;
-			event_copy.param.conn.private_data_len =
-			    event->param.conn.private_data_len;
+			size_t len = MIN(event->param.conn.private_data_len,
```

Comment:
Seems, it's impossible case in a normal situation. The private data is always `sizeof(struct fi_conn_priv_params)`, isn't it?
But to be safe here, I'm okay with it

Reply:
One would expect, but it's not what I saw. fi_conn_priv_params is 40 bytes, but every length I got was between 56 and 190 bytes. No idea why. Hence the MIN.

Reply:
> No idea why

Thanks for the explanation. Seems something is hidden by rdma cm library :)

Reply:
The underlying CMs always carry the full amount of private data available in the CM protocols.  For IB, this varies from 56 to 190 bytes depending on which message was sent.  There's no way to convey how much of what is received was initialized by the sender.